### ISSUE TITLE
TRANSIP: TXT-records incorrectly passed (with quotes)

### ISSUE THREAD

**Issue by cafferata** [StackExchange/dnscontrol#2702]
The DNS TXT-records are incorrectly passed to TransIP (with quotes).

```shell
dig jcid.nl. TXT
```
```diff
; <<>> DiG 9.10.6 <<>> jcid.nl. TXT
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 5537
;; flags: qr rd ra ad; QUERY: 1, ANSWER: 4, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 1232
;; QUESTION SECTION:
;jcid.nl.			IN	TXT

;; ANSWER SECTION:
jcid.nl.		86400	IN	TXT	"facebook-domain-verification=6uw144ekdcho9313m6s4tf50gbpxt8"
jcid.nl.		86400	IN	TXT	"pinterest-site-verification=1b9ccc530d0a8f711c7203e050d657a0"
jcid.nl.		86400	IN	TXT	"slack-domain-verification=8qbnweAyPOkmNkQyWcJQO2zWhWfPq0iWMX7YYFmC"
-jcid.nl.		86400	IN	TXT	"v=spf1 include:spf.jcid.nl include:_spf.google.com include:spf.mandrillapp.com include:_spf.exactonline.nl -all"
+jcid.nl.		86400	IN	TXT	"\"v=spf1 include:spf.jcid.nl include:_spf.google.com include:spf.mandrillapp.com include:_spf.exactonline.nl -all\""

;; Query time: 70 msec
;; SERVER: *******#53(*******)
;; WHEN: Fri Dec 08 19:56:15 CET 2023
;; MSG SIZE  rcvd: 384
```

Reproducible DNSControl config

```javascript
D('jcid.nl',
    PROVIDER_NONE,
    DnsProvider(PROVIDER_TRANSIP),
    SPF_BUILDER({
        label: '@',
        parts: [
            'v=spf1',
            '-all'
        ]
    })
);
```

![Screenshot 2023-12-08 at 20 05 59](https://github.com/StackExchange/dnscontrol/assets/1150425/3fc5652b-ea05-4da8-8a14-84acd3091810)

cc: @blackshadev

**Comment by tlimoncelli** [comment]
I'm the last to modify that code therefore this is probably my fault. My apologies.

I haven't found an automated way to detect this bug. However, I did write up a manual way to  test this here: 

https://docs.dnscontrol.org/developer-info/testing-txt-records

**Comment by cafferata** [comment]
ef081da1a6c8327ac1b27672f21a23c836b6ca7d / https://github.com/StackExchange/dnscontrol/pull/2673

```shell
git checkout ef081da1a6c8327ac1b27672f21a23c836b6ca7d
go build
```
```shell
dnscontrol preview --domains jcid.nl
```
```shell
******************** Domain: jcid.nl
Done. 0 corrections.
```

cbccbbeb8d981bcd688de1ee6ef8efe8df8a56d9 / https://github.com/StackExchange/dnscontrol/pull/2631

```shell
git checkout cbccbbeb8d981bcd688de1ee6ef8efe8df8a56d9
go build
```
```shell
dnscontrol preview --domains jcid.nl
```
```shell
******************** Domain: jcid.nl
6 corrections (transip)
#1: [2/2] delete: ± MODIFY jcid.nl TXT ("v=spf1include:spf.jcid.nlinclude:_spf.google.cominclude:spf.mandrillapp.cominclude:_spf.exactonline.nl-all" ttl=86400) -> ("v=spf1 include:spf.jcid.nl include:_spf.google.com include:spf.mandrillapp.com include:_spf.exactonline.nl -all" ttl=86400)
#2: [1/2] create: ± MODIFY jcid.nl TXT ("v=spf1include:spf.jcid.nlinclude:_spf.google.cominclude:spf.mandrillapp.cominclude:_spf.exactonline.nl-all" ttl=86400) -> ("v=spf1 include:spf.jcid.nl include:_spf.google.com include:spf.mandrillapp.com include:_spf.exactonline.nl -all" ttl=86400)
Done. 2 corrections.
```

**Comment by cafferata** [comment]
> However, I did write up a manual way to test this here:
> https://docs.dnscontrol.org/developer-info/testing-txt-records

Done! 

```shell
dnscontrol preview --domains jcid.nl
```
```shell
2023/12/08 23:01:50 printIR.go:107: 2 Validation errors:
2023/12/08 23:01:50 printIR.go:113: ERROR: TRANSIP rejects domain jcid.nl: txtstring contains doublequotes
2023/12/08 23:01:50 printIR.go:113: ERROR: TRANSIP rejects domain jcid.nl: txtstring contains doublequotes
exiting due to validation errors
```

**Comment by blackshadev** [comment]
Fixed in #2708
